core/app/models/spree/order.rb

```diff
@@ -105,6 +105,8 @@ def states
 
     # Returns
     has_many :return_authorizations, dependent: :destroy, inverse_of: :order
+    has_many :return_items, through: :inventory_units
+    has_many :customer_returns, through: :return_items
```

Comment:
Have you only added these for the tests? If so, I think you should move the logic in the tests. I like having these associations but I don't know if they belong to this PR. Also, I'm not 100% sure this is the right way to associate returns to orders, it would probably be much easier and less error-prone to do it directly as you have proposed. I think we need to be mindful of people doing weird stuff (e.g. moving inventory units between orders maybe? 😄).

Reply:
About this - I felt the way that customer returns were accessed from the admin side was too convoluted for what I wanted to write (https://github.com/solidusio/solidus/blob/1aa554fb04d2e28d279ec28a6ddb36b6ca2e5956/backend/app/controllers/spree/admin/customer_returns_controller.rb#L43) so I added the associations to be able to reference the customer_return from the order - inventory_units was the closest association I could find that has an order_id, but I had to use return_items as an intermediary connection. It's necessary for the API endpoint to be able to find the return_items for any given order.

I would much rather add an order_id to customer_returns (as I think they don't make sense outside the context of an order anyway) but I thought it might be a bit out of the scope of this PR (and I would want to update the way the backend route fetches the customer_returns as well, which is definitely out of the scope of this PR), so I was going to open an issue about it when this PR was merged... but if you'd prefer, I can definitely add the order_id to simplify this association.

Reply:
Didn't realize this was being used in the controller. I think the new association makes sense as a first step, and we can always add the new DB column later (which I still think is the best way to go).